syntax = "proto3";

option csharp_namespace = "Ocr.Grpc";
option java_multiple_files = true;
option java_package = "com.example.ocr";

package ocrworker;

service OcrWorker {
  rpc Ping (PingRequest) returns (PingResponse);
  rpc ExtractText (ExtractTextRequest) returns (ExtractTextResponse);
}

message PingRequest {
  string message = 1;
}

message PingResponse {
  string message = 1;
}

message ExtractTextRequest {
  string request_id = 1;
  string file_path = 2;
  string language = 3;
  bool use_angle_cls = 4;
}

message ExtractTextResponse {
  bool ok = 1;
  string text = 2;
  string error = 3;
  int64 elapsed_ms = 4;
}
